src/Ouroboros/Network/Chain.hs

```diff
@@ -273,6 +280,19 @@ selectPoints offsets =
       where
         c' = drop off c
 
+findBlock
+  :: (block -> Bool)
+  -> Chain block
+  -> Maybe block
+findBlock _ Genesis  = Nothing
```

Comment:
I was wondering about this back when this module was first introduced: is it (going to be) a problem that the genesis block here is not a first-class block? If one peer has only the genesis block in the chain, will it have trouble using what we have here? It will not be able to `findBlock` for any blocks that it has.

Reply:
I think the genesis block not being first class is the right thing, having thought about it before. Though it does have some corner cases (either way one does it). The `ChainFragment` rep that we'll be using in practice for many places also only covers "real" blocks.

In the use case in the fetch logic, we're using `findBlock` to find the first point on the chain that does have a block previously downloaded. If we were to hit the genesis block there, that's fine, we use:
```
chainFetchSuffix alreadyDownloaded (ChainSuffix chain p) =
    case findBlock (alreadyDownloaded . castPoint . blockPoint) chain of
      Nothing -> ChainSuffix chain p
      Just b  -> ChainSuffix chain (blockPoint b)
```
So the range that we'd be asking for in the `Nothing` case would be all blocks from (but not including) the genesis point, up to some later point.